src/redux/slices/persistGateHydration.ts

```diff
@@ -0,0 +1,56 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { REHYDRATE } from 'redux-persist';
+
+import { RootState } from '../RootState';
+
+import SliceName from '@/redux/types/SliceName';
+
+export type PersistGateHydrationState = {
+  isPersistGateHydrationInProgress: boolean;
+  isPersistGateHydrationComplete: boolean;
+};
+
+const initialState: PersistGateHydrationState = {
+  isPersistGateHydrationInProgress: false,
+  isPersistGateHydrationComplete: false,
+};
+
+// This slice checks if redux-persist has finished hydrating the store.
+// Hydration happens a bit after the page loads, useful for preventing
+// the animations until the state is settled in.
+export const persistGateHydrationSlice = createSlice({
+  name: SliceName.PERSIST_GATE_HYDRATION,
+  initialState,
+  reducers: {
+    setIsPersistGateHydrationInProgress: (
+      state: PersistGateHydrationState,
+      action: PayloadAction<boolean>,
+    ) => ({
+      ...state,
+      isPersistGateHydrationInProgress: action.payload,
+    }),
+    setIsPersistGateHydrationComplete: (
+      state: PersistGateHydrationState,
+      action: PayloadAction<boolean>,
+    ) => ({
+      ...state,
+      isPersistGateHydrationComplete: action.payload,
+    }),
+  },
+  extraReducers: (builder) => {
+    builder.addCase(REHYDRATE, (state: PersistGateHydrationState) => {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
I think the reducer should not directly manipulate `state` but rather return a new object e.g.:

```
{
      ...state,
      isPersistGateHydrationInProgress: action.payload,
    }
```

You can refer to other instances of `builder.addCase` for example:

<img width="837" alt="Screen Shot 2022-12-27 at 11 45 08 AM" src="https://user-images.githubusercontent.com/15169499/209612403-5865a018-444c-4e76-9231-8adee935edd7.png">